File: easy/rightmost.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    my ( $n, $m ) = ( $line =~ /([^,]+),([^\n])/ );
    printf( "%d\n", rindex( $n, $m ) );
}
close(INFILE);

File: easy/rightmost.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp
  next if s.empty?
  m = n = -1
  loop do
    m = n
    n = s.index(s[-1], n + 1)
    break if n == s.length - 1
  end
  puts m
end
